### DISCUSSION TITLE
How to encode a path parameter?

### DISCUSSION THREAD

**Discussion by fauxbytes** [curl/curl#10678] [Q&A]
Hi.
Curl supports encoding of query params via `--data-urlencode`. Is there some option to encode path parameters?

E.G: to query the endpoint `api.foo.com/user/by_name/:namestr` with "Jane Doe" as the value of passed in place of `:namestr`? Is sth like `curl -G https://api.foo.com/user/by_name/ --path-param-encode="Jane Doe"` to produce the reqeust url: 
```
https://api.foo.com/user/by_name/Jane%20Dow
```
?
Thanks!

**Comment by andy5995** [comment]
It looks like you're not the only one who's interested in that functionality:

https://stackoverflow.com/questions/296536/how-to-urlencode-data-for-curl-command

https://unix.stackexchange.com/questions/86729/any-way-to-encode-the-url-in-curl-command

Here's something that you might be able to use until you get a real answer:

`curl https://api.foo.com/user/by_name/$(printf "Jane Doe" | jq -sRr '@uri')`
